subsys/net/ip/connection.c

```diff
@@ -471,42 +478,44 @@ static inline void conn_send_icmp_error(struct net_pkt *pkt)
 	}
 }
 
-static bool conn_are_end_points_valid(struct net_pkt *pkt,
-				      union net_ip_header *ip_hdr,
-				      uint16_t src_port,
-				      uint16_t dst_port)
+#if defined(CONFIG_NET_IP)
+static bool conn_are_endpoints_valid(struct net_pkt *pkt, uint8_t family,
+				     union net_ip_header *ip_hdr,
+				     uint16_t src_port, uint16_t dst_port)
 {
-	bool my_src_addr = false;
-
-	/* For AF_PACKET family, we are not parsing headers. */
-	if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) &&
-	    net_pkt_family(pkt) == AF_PACKET) {
+	bool is_my_src_addr;
+	bool is_same_src_and_dst_addr;
+
+	switch (family) {
+#if defined(CONFIG_NET_IPV4)
+	case AF_INET:
+		is_my_src_addr = net_ipv4_is_my_addr(
+			(struct in_addr *)ip_hdr->ipv4->src);
+		is_same_src_and_dst_addr = net_ipv4_addr_cmp_raw(
+			ip_hdr->ipv4->src, ip_hdr->ipv4->dst);
+		break;
+#endif
+#if defined(CONFIG_NET_IPV6)
+	case AF_INET6:
+		is_my_src_addr = net_ipv6_is_my_addr(
+			(struct in6_addr *)ip_hdr->ipv6->src);
+		is_same_src_and_dst_addr = net_ipv6_addr_cmp_raw(
+			ip_hdr->ipv6->src, ip_hdr->ipv6->dst);
+		break;
+#endif
+	default:
 		return true;
 	}
```

Comment:
We have been trying to put code info blocks that can be compile checked i.e., so using the `IS_ENABLED()` macro when possible. We should be able to still use that macro here like in the original code.

Reply:
@jukkar First of all, thank you so much for taking the time to review this stuff. And yeah, I understand why you prefer IS_ENABLED() - to be honest I didn't fully understand the rationale behind it until now. I did this to prepare for factoring this stuff out to separate files in the future. But of course this can (and should) be done then. There is a trade off to IS_ENABLED() but taking into account what you're saying I understand what the advantages are and will revert as many of these instances as possible. So: good point!

Reply:
Fixed everywhere.